src/libYARP_OS/src/Node.cpp

```diff
@@ -202,7 +230,7 @@ class yarp::os::Node::Helper : public PortReader
     void add(Contactable& contactable);
     void update(Contactable& contactable);
     void remove(Contactable& contactable);
-    Contact query(const ConstString& name, const ConstString& category);
+    std::vector<Contact> query(const ConstString& name, const ConstString& category = ConstString());
 
     void prepare(const ConstString& name);
```

Comment:
@drdanz I think we should move the `mutex.lock()` and `mutex.unlock()` also in the helper class for the `prepare()` method, just to be consistent.